### PR TITLE
NO-ISSUE: Make sure `kie-tools` is using the latest timestamped SNAPSHOT from Kogito (999-20240411-SNAPSHOT)

### PR DESCRIPTION
**Pull request by yesamer**
Update all Kogito references to `999-20240411-SNAPSHOT`

### CODE REVIEW

File: packages/stunner-editors/pom.xml [hunk 1]
```diff
@@ -326,7 +326,7 @@
     <version.zanata.plugin>2.3.0</version.zanata.plugin>
 
     <!-- KIE Libraries -->
-    <version.org.kie>999-20240331-SNAPSHOT</version.org.kie>
+    <version.org.kie>999-20240407-SNAPSHOT</version.org.kie>
```
Comment (tiagobento) [inline packages/stunner-editors/pom.xml]: Can we start using the version from `build-env` here? Adding an `install.js` script etc.

File: packages/kn-plugin-workflow/env/index.js [hunk 2]
```diff
@@ -39,12 +39,12 @@ module.exports = composeEnv([require("@kie-tools/root-env/env")], {
...
-      default: "quay.io/kiegroup/kogito-swf-devmode-nightly:main-2024-04-03",
+      default: "quay.io/kiegroup/kogito-swf-devmode-nightly:main-2024-04-05",
       description: "SonataFlow dev mode image (used on cli run)",
     },
     KN_PLUGIN_WORKFLOW__kogitoVersion: {
       name: "KN_PLUGIN_WORKFLOW__kogitoVersion",
-      default: "999-20240331-SNAPSHOT",
+      default: "999-20240407-SNAPSHOT",
       description: "Kogito version to be used when creating and converting to Quarkus Projects",
     },
```
Comment (tiagobento) [inline packages/kn-plugin-workflow/env/index.js]: I think we can remove this env and start using `kogitoRuntime.version` from root-env.. WDYT? cc @paulovmr @caponetto

**Review by tiagobento — CHANGES_REQUESTED** [review]
I noticed we have different dates for the images... Could we unify?

**Comment by yesamer** [comment]
@tiagobento At this time, the latest version of the image https://quay.io/repository/kiegroup/kogito-base-builder-nightly?tab=tags&tag=latest is 2024-04-05

File: packages/stunner-editors/pom.xml (hunk 1 above)
Comment (yesamer) [reply packages/stunner-editors/pom.xml]: I guess yes, good idea.

**Comment by tiagobento** [comment]
@ricardozanini Any idea what could've happened with the nightly build of this image?

File: packages/stunner-editors/pom.xml (hunk 1 above)
Comment (yesamer) [reply packages/stunner-editors/pom.xml]: Done :)

File: packages/kn-plugin-workflow/env/index.js (hunk 2 above)
Comment (paulovmr) [reply packages/kn-plugin-workflow/env/index.js]: Sure, makes sense.

**Comment by ricardozanini** [comment]
@tiagobento I'll take a look on Jenkin

**Comment by ricardozanini** [comment]
@yesamer @tiagobento this image has been removed as part of the old operator archiving: https://github.com/apache/incubator-kie-kogito-images/pull/1748

**Comment by ricardozanini** [comment]
If you want, I think I can help review these PRs related to images. Let me know!

**Comment by yesamer** [comment]
@tiagobento @thiagoelg That means we need a replacement for that... Should I open a ticket?

**Comment by ricardozanini** [comment]
@yesamer why do you need a replacement? That image won't be used anywhere as far as I know.

**Comment by tiagobento** [comment]
@ricardozanini It is used by Serverless Logic Web Tools, it seems like... cc @caponetto

**Comment by paulovmr** [comment]
@ricardozanini @tiagobento yes this image was the base for the web tools builder and dev-mode images. @ricardozanini could it be restored?

**Comment by ricardozanini** [comment]
@paulovmr yes, we can. I can't do it today, if some of you have some room, here's the PR where I removed it: https://github.com/apache/incubator-kie-kogito-images/pull/1748/files

**Comment by ricardozanini** [comment]
@paulovmr @caponetto honestly, I saw the image now and I'm not seeing anything that it's not on openjdk-17 images. Why not use it directly?

**Comment by ricardozanini** [comment]
Here: https://github.com/apache/incubator-kie-kogito-images/pull/1754

**Comment by rodrigonull** [comment]
@yesamer I'm running the weekly jobs to generate new timestamped snapshots. We should have the 999-20240411-SNAPSHOTs available by tomorrow morning.

**Comment by yesamer** [comment]
@rodrigonull Great!!!! Are you launching the job for all our repos? jit-executor as well? Thank you!

**Comment by rodrigonull** [comment]
@yesamer yes, my plan is to trigger all jobs including the JIT executor.

**Comment by tiagobento** [comment]
@yesamer For images, please change the tags to 999-YYYYMMDD instead of main-*, as the first comes from the weekly jobs (dependencies are also timestamped SNAPSHOTs) and the second is from daily jobs (dependencies are 999-SNAPSHOT)

**Comment by yesamer** [comment]
@tiagobento Done!

**Review by tiagobento — APPROVED** [review]

**Review by gitgabrio — COMMENTED** [review]
Hi @yesamer 
Out of curiosity: would not be possible to store that variable in one single place, and maybe refer to it wherever is needed ?

**Comment by tiagobento** [comment]
@gitgabrio That's the idea behind https://github.com/apache/incubator-kie-tools/blob/main/packages/root-env/env/index.js#L53-L60, but it's not always simple to do it, especially when the format changes slightly :)

**Comment by yesamer** [comment]
@gitgabrio I guess @tiagobento can give you a better answer than me, anyway my feeling is that is it possible to remove some of those places, and the tooling team is already committed to that. In this PR I removed one place :)

**Comment by yesamer** [comment]
@tiagobento Can I merge it? :)

**Comment by tiagobento** [comment]
@yesamer There's only one approve :( We need 2.

In the mean time...... could we remove `KN_PLUGIN_WORKFLOW__quarkusVersion` and `KN_PLUGIN_WORKFLOW__kogitoVersion`? We can use the one from `root-env` instead of having two variables dedicated to the `kn-plugin-workflow`.. and we @paulovmr's blessing :)

**Comment by yesamer** [comment]
@tiagobento Right.

So should I just remove KN_PLUGIN_WORKFLOW__kogitoVersion and replace it with `kogitoRuntime.version`, correct?

**Review by gitgabrio — APPROVED** [review]

**Comment by tiagobento** [comment]
@yesamer Yes!

**Review by klbowers — APPROVED** [review]

**Comment by tiagobento** [comment]
@yesamer Looks like this one can be removed too! `SERVERLESS_LOGIC_WEB_TOOLS__kogitoVersion` cc @paulovmr

**Comment by tiagobento** [comment]
@yesamer And this one too! `SERVERLESS_LOGIC_WEB_TOOLS__quarkusPlatformVersion`.

This is the last comment I promise 😅

**Comment by tiagobento** [comment]
I bet @gitgabrio is thrilled with us removing env vars, though :)

**Comment by yesamer** [comment]
BINGO!

**Comment by yesamer** [comment]
Done! @tiagobento @gitgabrio @kbowers-ibm

**Pull request merged**